1_fetch/src/get_nwis_data.R

```diff
@@ -1,15 +1,18 @@
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500")){
-  
-  # create the file names that are needed for download_nwis_site_data
-  # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
-  # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
+make_table <- function(site_data_dir){
+  file_names = dir(site_data_dir)
+  m5sums = as.vector(md5sum(site_data_dir))
+  df = data.frame('filename' = file_names, 'm5sums'=m5sums)
+  return(df)
+}
+
+
+combine_csvs <- function(input_data_table, out_dir){
+  # loop through downloaded files
+  download_files <- file.path(out_dir, input_data_table$filename)
   data_out <- data.frame(agency_cd = c(), site_no = c(), dateTime = c(), 
                          X_00010_00000 = c(), X_00010_00000_cd = c(), tz_cd = c())
-  # loop through files to download 
   for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
     # read the downloaded data and append it to the existing data.frame
     these_data <- read_csv(download_file, col_types = 'ccTdcc')
     data_out <- rbind(data_out, these_data)
```

Comment:
This `for` loop works but might be a bit slower than
```r
data_out <- purrr::map_df(download_files, function(download_file) {
  read_csv(download_file, col_types = 'ccTdcc')
}
```
which I believe is all you'd need to get the data read and combined (with some possible debugging needed - I didn't test this code). No need to initialize `data_out` beforehand unless you expect cases where the download_files are all empty or there aren't any of them.

Reply:
I guess this is actually a comment for @jread-usgs now that I understand what I'm reviewing =)

Reply:
Is _your_ comment for me or Jeff's question?

Reply:
_My_ comment is for you. Recommending `purrr` rather than an iterative `rbind`.

Reply:
yeah, that's a lot cleaner and better. I think at the time I was focusing on some simpler code, and for loops seem to be something that everybody gets what's going on. But I don't think I was consistent with coding it with those objectives...so probably a mix of easy to read vs other.